src/main/java/seedu/address/logic/util/LessonModificationUtil.java

```diff
@@ -0,0 +1,110 @@
+package seedu.address.logic.util;
+
+import static seedu.address.logic.commands.AddAttendanceCommand.MESSAGE_DUPLICATE_ATTENDANCE;
+import static seedu.address.logic.commands.DeleteAttendanceCommand.MESSAGE_MISSING_ATTENDANCE;
+
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.attendance.Attendance;
+import seedu.address.model.attendance.AttendanceRecord;
+import seedu.address.model.attendance.AttendanceRecordList;
+import seedu.address.model.attendance.Week;
+import seedu.address.model.lesson.Day;
+import seedu.address.model.lesson.Lesson;
+import seedu.address.model.lesson.NumberOfOccurrences;
+import seedu.address.model.lesson.Venue;
+import seedu.address.model.student.Student;
+
+/**
+ * Contains utility methods for modifying {@code Attendances} in {@code Lessons}.
+ */
+public class LessonModificationUtil {
```

Comment:
Nice job extracting these methods to a separate class.

Since they are public methods now, perhaps we should have unit tests for them? It can come in a separate PR subsequently though.

Reply:
Lets do it in a separate PR.